### ISSUE TITLE
Shortcodes does not work correctly when timeframe spans between days

### ISSUE THREAD

**Issue by tripflex** [cliffordp/tk-event-weather#9]
Using ISO 8601 timestamps in shortcode:

```
[tk_event_weather  lat_long='28.5549259,-81.3342398' start_time='2016-08-28T02:00:00+00:00' end_time='2016-08-28T06:30:00+00:00']
```

**For reference:**
`2016-08-28T02:00:00+00:00` = RFC 2822 time `Sat, 27 Aug 2016 22:00:00 -04:00`
`2016-08-28T06:30:00+00:00` = RFC 2822 time `Sun, 28 Aug 2016 02:30:00 -04:00`

Displays this:
![](https://smyl.es/img/Screenshot_2016-08-19_08-49-56_PM.png)

Notice it's missing the majority of the forecasts ... my guess is a there's a bug with anything that spans into the next day, because any timeframes during the same day output fine.

**Comment by cliffordp** [comment]
Hi again @tripflex.

This is the way it works on purpose.

The Forecast.io API currently requires a separate API call per day and I decided to limit it to a single day to avoid unknowing quantities of API calls (obviously not for someone like yourself _trying_ to do this actual thing).

The plugin's byline is:

> Display beautiful, accurate, and free weather forecasts between a start and end time on **_the same day._** Perfect for event calendars. (emphasis added)

Forecast.io told me that they will eventually release a "bulk data" API to pull data for multiple days in a single API call. Until that comes out...

I believe the issue happening here is that Forecast.io sets the timezone based on the location. Since the location is in Orlando, FL (New York timezone), it's correctly displaying 10pm Orlando time but, due to the limitations of the API, it only works for 1 day at a time.

You might want to try something like this (see following comment about using `darksky_credit_link_off` instead):

```
[tk_event_weather lat_long='28.5549259,-81.3342398' start_time='2016-08-27T22:00:00-04:00' end_time='2016-08-27T23:59:00-04:00' class='tkeventw-myclass']

[tk_event_weather lat_long='28.5549259,-81.3342398' start_time='2016-08-28T00:00:00-04:00' end_time='2016-08-28T04:30:00-04:00' class='tkeventw-myclass' forecast_io_credit_link_off='true']
```

And add some CSS like this:

```
.tkeventw-myclass {
    display: inline-block;
    vertical-align: top;
}
```

This is how it turned out for me:
<img width="781" alt="image 2016-08-24 at 11 38 33 am" src="https://cloud.githubusercontent.com/assets/1812179/17939249/67249728-69ef-11e6-93ef-b77a562ad978.png">

Note: the `class` shortcode argument was there from the beginning but wasn't implemented properly (i.e. a bug). I've fixed that bug in version 1.2.3.3 (released a few minutes ago).

Please let me know how things go for you.

**Comment by cliffordp** [comment]
Note that, as of version 1.2.4 (October 30, 2016) and more recent, the above snippet's use of `forecast_io_credit_link_off` should be replaced with `darksky_credit_link_off` to get the same effect.

**Comment by cliffordp** [comment]
Similar request: 7-10 day forecast: https://wordpress.org/support/topic/hourly-weather/#post-8862598

**Comment by cliffordp** [comment]
This will be available in TK Event Weather version 1.5, to be released soon.
